lib/libzfs/libzfs_pool.c

```diff
@@ -4171,6 +4148,32 @@ zpool_label_disk_check(char *path)
 }
 
 /*
+ * Generate a unique partition name for the ZFS member.  Partitions must
+ * have unique names to ensure udev will be able to create symlinks under
+ * /dev/disk/by-partlabel/ for all pool members.  The partition names are
+ * of the form <pool>-<unique-id>.
+ */
+static void
+zpool_label_name(char *label_name, int label_size)
+{
+	uint64_t id = 0;
+	int fd;
+
+	fd = open("/dev/urandom", O_RDONLY);
+	if (fd > 0) {
+		if (read(fd, &id, sizeof (id)) != sizeof (id))
+			id = 0;
+
+		close(fd);
+	}
+
+	if (id == 0)
+		id = (((uint64_t)rand()) << 32) | (uint64_t)rand();
+
+	sprintf(label_name, "zfs-%016llx", (u_longlong_t) id);
```

Comment:
That would work for `zpool create`, but the trouble is it needs to be a persistent unique ID.  We need to avoid the case where a pool ends up with two partitions using the same name.  You could see this happening quickly if every `zpool replace` ended up generating `zfs-0` as the partition name.